test/linter/test-browsers-presence.js

```diff
@@ -39,16 +25,46 @@ const browsers = {
  * @param {string} [path] The path of the data.
  * @returns {void}
  */
-function processData(
-  data,
-  displayBrowsers,
-  requiredBrowsers,
-  category,
-  logger,
-  path = '',
-) {
+function processData(data, category, logger, path = '') {
   if (data.__compat && data.__compat.support) {
     const support = data.__compat.support;
+    const definedBrowsers = Object.keys(support);
+
+    let displayBrowsers = [
+      ...Object.keys(browsers).filter((b) =>
+        ['desktop', 'mobile'].includes(browsers[b].type),
+      ),
+      ,
+    ];
+    let requiredBrowsers = [
+      ...Object.keys(browsers).filter((b) => browsers[b].type == 'desktop'),
+    ];
```

Comment:
What's the advantage of spreading the `Array.filter()` result into a fresh array?

```suggestion
    let displayBrowsers = Object.keys(browsers).filter((b) =>
        ['desktop', 'mobile'].includes(browsers[b].type));
    let requiredBrowsers = Object.keys(browsers).filter((b) => browsers[b].type == 'desktop');
```